src/vm/contexts.rs

```diff
@@ -890,7 +937,17 @@ impl<'a, 'b> Environment<'a, 'b> {
             match res {
                 Ok(value) => {
                     let sender_principal = self.sender.clone().map(|v| v.expect_principal());
-                    handle_contract_call_special_cases(&mut self.global_context, sender_principal.as_ref(), contract_identifier, tx_name, &value)?;
+                    let sponsor_principal = match self.sponsor.clone() {
```

Comment:
Minor nitpick -- you have a lot of dangling whitespace here.  I guess `cargo fmt` didn't catch it?

Reply:
I ran `cargo fmt`, it didn't catch any issues here?